lib/Dialect/TensorExt/IR/TensorExtAttributes.td

```diff
@@ -64,22 +66,142 @@ def SIMDPacking_Attr : TensorExt_Attr<"SIMDPacking", "simd_packing",
   let assemblyFormat =  "`<` struct(params) `>`";
 }
 
-def OriginalType_Attr : TensorExt_Attr<"OriginalType", "original_type"> {
+def TensorExt_AlignmentAttr : TensorExt_Attr<"Alignment", "alignment"> {
+  let summary = "An attribute describing padding and alignment of a tensor.";
+  let description = [{
+    This attribute is used to describe how a data-semantic tensor is padded
+    and replicated to align its size before applying a ciphertext layout
+    (see `tensor_ext.layout`).
+
+    It describes transformations to be applied to an input tensor.
+
+    The `in` attribute describes the shape of the original tensor.
+    The following transformations are applied to the input tensor.
+
+    1. New unit dimensions are inserted to match the number of dimensions
+       of the output tensor. The `insertedDims` attribute is an array
+       specifying which axes in the output tensor are new unit dimensions. By
+       default no new dimensions are inserted.
+    2. Padding is applied. The `padding` attribute is an array with the same
+       size as the output tensor shape, which each entry denoting the number of
+       values to pad at the end of that axis. The `paddingValue` attribute,
+       defaulting to zero, describes the value used for padding.
```

Comment:
Update here if you go the route of a possible unspecified padding - The paddingValue attr describes the value used for padding. If left unspecified, then the program will allow any value to be present in the padded slots.

Reply:
I went the route of forcing a padding value whenever padding is set.